File: v-lang/lang.java/Spring/web/src/main/java/org/example/controller/advice/ExceptionAdvice.java
package org.example.controller.advice;

import lombok.extern.slf4j.Slf4j;
import org.example.exception.BizException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import java.util.HashMap;
import java.util.Map;

@Slf4j
@RestControllerAdvice
public class ExceptionAdvice {

    @ExceptionHandler(BizException.class)
    public ResponseEntity<Map<String, Object>> handleBizException(BizException e) {
        log.error("ExceptionAdvice", e);
        Map<String, Object> body = new HashMap<>();
        body.put("msg", e.getMessage());
        return new ResponseEntity<>(body, e.getHttpStatus());
    }

    @ExceptionHandler(Exception.class)
    public ResponseEntity<Map<String, Object>> handleException(Exception e) {
        log.error("ExceptionAdvice", e);
        Map<String, Object> body = new HashMap<>();
        body.put("msg", "服务器错误");
        return new ResponseEntity<>(body, HttpStatus.INTERNAL_SERVER_ERROR);
    }
}

File: v-demo/demo-fs/fs-spring/src/main/kotlin/org/example/controller/advice/ExceptionAdvice.kt
package org.example.controller.advice

import io.github.oshai.kotlinlogging.KotlinLogging
import org.example.BizException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class ExceptionAdvice {
    private val logger = KotlinLogging.logger {}

    @ExceptionHandler(BizException::class)
    fun handleBizException(e: BizException): ResponseEntity<BizErrorResult> {
        logger.error(e) { "ExceptionAdvice ${e.message}" }
        val bizErrorResult = BizErrorResult(message = e.message)
        return ResponseEntity(bizErrorResult, e.httpStatus)
    }

    @ExceptionHandler(Exception::class)
    fun handleException(e: Exception): ResponseEntity<BizErrorResult> {
        val bizErrorResult = BizErrorResult()
        logger.error(e) { "ExceptionAdvice" }
        return ResponseEntity(bizErrorResult, HttpStatus.INTERNAL_SERVER_ERROR)
    }
}

data class BizErrorResult(val message: String? = "服务器错误")